src/core_modules/capture-core/rules/engine/converters/inputValueConverter.js

```diff
@@ -51,28 +52,36 @@ class RulesValueConverter implements IConvertInputRulesValue {
         return value || '';
     }
 
-    convertNumber(value: any): number | string {
-        return (value || value === 0) ? value : '';
+    convertNumber(value: number | ''): number {
+        return value || 0;
     }
 
-    convertInteger(value: any): number | string {
-        return (value || value === 0) ? value : '';
+    convertInteger(value: number | ''): number {
+        return value || 0;
     }
 
-    convertIntegerPositive(value: any): number | string {
-        return (value || value === 0) ? value : '';
+    convertIntegerPositive(value: number | ''): number {
+        return value || 0;
     }
 
-    convertIntegerNegative(value: any): number | string {
-        return (value || value === 0) ? value : '';
+    convertIntegerNegative(value: number | ''): number {
+        return value || 0;
     }
 
-    convertIntegerZeroOrPositive(value: any): number | string {
-        return (value || value === 0) ? value : '';
+    convertIntegerZeroOrPositive(value: number | ''): number {
+        return value || 0;
     }
 
-    convertPercentage(value: any): number | string {
-        return (value || value === 0) ? value : '';
+    convertPercentage(value: string): number {
+        if (!value) {
+            return 0;
+        }
+        const numberValue = parseNumber(value);
+        if (isNaN(numberValue)) {
+            return 0;
+        }
+
+        return numberValue / 100;
```

Comment:
Wondering if we wanna define a fixed amount of decimals here. to avoid cases like `3.213123%`

[toFixed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed ) could be a quick way to do this.

Reply:
I think, since this value will be used in calculations, we shouldn't manipulate it. We should rather supply a d2 helper function to be used in the rules if needed (in the same vein as d2:ceil, d2:floor, d2:round).